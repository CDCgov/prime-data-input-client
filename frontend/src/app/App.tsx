import React, { useEffect, useState } from "react";
import { gql, useQuery } from "@apollo/client";
import { ToastContainer } from "react-toastify";
import { useDispatch, connect } from "react-redux";
import "react-toastify/dist/ReactToastify.css";
import { Redirect, Route, Switch } from "react-router-dom";
import { AppInsightsContext } from "@microsoft/applicationinsights-react-js";
import { reactPlugin } from "./AppInsights";

import PrimeErrorBoundary from "./PrimeErrorBoundary";
import Header from "./commonComponents/Header";
import USAGovBanner from "./commonComponents/USAGovBanner";
import LoginView from "./LoginView";
import { setInitialState } from "./store";
import TestResultsListContainer from "./testResults/TestResultsListContainer";
import TestQueueContainer from "./testQueue/TestQueueContainer";
import ManagePatientsContainer from "./patients/ManagePatientsContainer";
import EditPatientContainer from "./patients/EditPatientContainer";
import AddPatient from "./patients/AddPatient";
import ManageOrganizationContainer from "./Settings/ManageOrganizationContainer";
import ManageFacilitiesContainer from "./Settings/Facility/ManageFacilitiesContainer";
import FacilityFormContainer from "./Settings/Facility/FacilityFormContainer";
import { getFacilityIdFromUrl } from "./utils/url";

const WHOAMI_QUERY = gql`
  query WhoAmI {
    whoami {
      id
      firstName
      middleName
      lastName
      suffix
      organization {
        name
        testingFacility {
          id
          name
        }
      }
    }
  }
`;

const SettingsRoutes = ({ match }: any) => (
  <>
    {/* note the addition of the exact property here */}
    <Route exact path={match.url} component={ManageOrganizationContainer} />
    <Route
      path={match.url + "/facilities"}
      component={ManageFacilitiesContainer}
    />
    <Route
      path={match.url + "/facility/:facilityId"}
      render={({ match }) => (
        <FacilityFormContainer facilityId={match.params.facilityId} />
      )}
    />
    <Route
      path={match.url + "/add-facility/"}
      render={({ match }) => (
        <FacilityFormContainer facilityId={match.params.facilityId} />
      )}
    />
  </>
);

const App = () => {
  const dispatch = useDispatch();
  const { data, loading, error } = useQuery(WHOAMI_QUERY, {
    fetchPolicy: "no-cache",
  });
  const [facilityId, updateFacilityId] = useState<string | null>("");

  useEffect(() => {
    if (!data) return;

    const getDefaultFacilityId = () => {
      const queryParamsFacilityId = getFacilityIdFromUrl();
      if (queryParamsFacilityId) {
        return queryParamsFacilityId;
      }

      const tucsonMountains = data.whoami.organization.testingFacility.find(
        (f: Facility) => f.name === "Tucson Mountains"
      );
      if (tucsonMountains) {
        return tucsonMountains.id;
      }

      return data.whoami.organization.testingFacility[0].id;
    };

    // if facilityId is not a query param, we need to refresh the page and add it
    if (!getFacilityIdFromUrl()) {
      window.location.href = `${
        window.location.pathname
      }?facility=${getDefaultFacilityId()}`;
    }

    const getDefaultFacility = () => {
      let facilityId = getDefaultFacilityId();
      return data.whoami.organization.testingFacility.find(
        (f: Facility) => f.id === facilityId
      );
    };

    dispatch(
      setInitialState({
        organization: {
          name: data.whoami.organization.name,
        },
        facilities: data.whoami.organization.testingFacility,
        facility: getDefaultFacility(),
        user: {
          id: data.whoami.id,
          firstName: data.whoami.firstName,
          middleName: data.whoami.middleName,
          lastName: data.whoami.lastName,
          suffix: data.whoami.suffix,
        },
      })
    );

    updateFacilityId(getFacilityIdFromUrl());
    // eslint-disable-next-line
  }, [data]);

  if (loading) {
    return <p>Loading account information...</p>;
  }

  if (error) {
    throw error;
  }

  return (
    <AppInsightsContext.Provider value={reactPlugin}>
      <PrimeErrorBoundary
        onError={(error: any) => (
          <div>
            <h1> There was an error. Please try refreshing</h1>
            <pre> {JSON.stringify(error, null, 2)} </pre>
          </div>
        )}
      >
        <div className="App">
          <div id="main-wrapper">
            <USAGovBanner />
            <Header facilityId={facilityId} />
            <Switch>
              <Route path="/login" component={LoginView} />
              <Route
                path="/queue"
                render={() => {
                  return <TestQueueContainer />;
                }}
              />
              <Route
                path="/"
                render={() => {
                  return <Redirect to="/queue" />;
                }}
                exact
              />
              <Route
                path="/results"
                render={() => {
                  return <TestResultsListContainer />;
                }}
              />
              <Route
                path={`/patients`}
                render={() => {
                  return <ManagePatientsContainer />;
                }}
              />
              <Route
                path={`/patient/:patientId`}
                render={({ match }) => (
                  <EditPatientContainer patientId={match.params.patientId} />
                )}
              />
              <Route path={`/add-patient/`} render={() => <AddPatient />} />
              <Route path="/settings" component={SettingsRoutes} />
            </Switch>
            <ToastContainer
              autoClose={5000}
              closeButton={false}
              limit={2}
              position="bottom-center"
              hideProgressBar={true}
            />
          </div>
        </div>
      </PrimeErrorBoundary>
    </AppInsightsContext.Provider>
  );
};

export default connect()(App);
