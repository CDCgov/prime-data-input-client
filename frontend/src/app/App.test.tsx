import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import createMockStore from "redux-mock-store";
import { MockedProvider } from "@apollo/client/testing";

import App, { WHOAMI_QUERY } from "./App";

jest.mock("uuid");
jest.mock("@microsoft/applicationinsights-react-js", () => ({
  ReactPlugin: jest.fn(),
}));

const mockStore = createMockStore([]);
const store = mockStore({});
const mocks = [
  {
    request: {
      query: WHOAMI_QUERY,
    },
    result: {
      data: {
        queue: [],
        organization: {
          testingFacility: [
            {
              id: "fec4de56-f4cc-4c61-b3d5-76869ca71296",
              deviceTypes: [
                {
                  internalId: "d70bb3b8-96bd-40d9-a3ce-b266a7edb91d",
                  name: "Quidel Sofia 2",
                  model: "Sofia 2 SARS Antigen FIA",
                  testLength: 15,
                },
                {
                  internalId: "5e44dcef-8cc6-44f4-a200-a5b8169ab60a",
                  name: "LumiraDX",
                  model: "LumiraDx SARS-CoV-2 Ag Test*",
                  testLength: 15,
                },
              ],
              defaultDeviceType: {
                internalId: "5e44dcef-8cc6-44f4-a200-a5b8169ab60a",
                name: "LumiraDX",
                model: "LumiraDx SARS-CoV-2 Ag Test*",
                testLength: 15,
              },
            },
          ],
        },
      },
    },
  },
];
describe("TestResultInputForm", () => {
  it("should render without error", () => {
    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Provider store={store}>
          <App />
        </Provider>
      </MockedProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
