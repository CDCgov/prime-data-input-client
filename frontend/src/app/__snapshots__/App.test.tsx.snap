// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render facility loading 1`] = `
<p>
  Loading facility information...
</p>
`;

exports[`App Render first loading screen 1`] = `
<p>
  Loading account information...
</p>
`;

exports[`App Render main screen 1`] = `
<div
  className="App"
>
  <div
    id="main-wrapper"
  >
    <div
      className="usa-banner"
    >
      <div
        className="usa-accordion"
      >
        <header
          className="usa-banner__header"
        >
          <div
            className="usa-banner__inner"
          >
            <div
              className="grid-col-auto"
            >
              <img
                alt="U.S. flag"
                className="usa-banner__header-flag"
                src="us_flag_small.png"
              />
            </div>
            <div
              className="grid-col-fill tablet:grid-col-auto"
            >
              <p
                className="usa-banner__header-text"
              >
                An official website of the United States government
              </p>
              <p
                aria-hidden="true"
                className="usa-banner__header-action"
              >
                Here’s how you know
              </p>
            </div>
            <button
              aria-controls="gov-banner"
              aria-expanded={false}
              className="usa-accordion__button usa-banner__button"
              onClick={[Function]}
            >
              <span
                className="usa-banner__button-text"
              >
                Here’s how you know
              </span>
            </button>
          </div>
        </header>
      </div>
    </div>
    <header
      className="usa-header usa-header--basic"
    >
      <div
        className="usa-nav-container"
      >
        <div
          className="usa-navbar"
        >
          <div
            className="usa-logo"
            id="basic-logo"
          >
            <a
              aria-current="page"
              aria-label="Home"
              className="active"
              href="/queue?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
              onClick={[Function]}
              style={Object {}}
              title="Home"
            >
              <img
                alt="{process.env.REACT_APP_TITLE}"
                className="width-card desktop:width-full"
                src="simplereport-logo-color.svg"
              />
            </a>
            <div
              className="prime-organization-name"
            >
              Organization Name
            </div>
          </div>
          <button
            className="usa-menu-btn"
            onClick={[Function]}
          >
            Menu
          </button>
        </div>
        <nav
          aria-label="Primary navigation"
          className="usa-nav prime-nav desktop:display-none"
        >
          <button
            className="fa-layers fa-fw fa-2x usa-nav__close prime-nav-close-button"
            onClick={[Function]}
            title="close menu"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-window-close fa-w-16 "
              data-icon="window-close"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
          <ul
            className="usa-nav__primary usa-accordion"
          >
            <li
              className="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden"
            >
              <a
                aria-current="page"
                className="prime-nav-link active-nav-item"
                href="/queue?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Conduct Tests
              </a>
            </li>
            <li
              className="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden"
            >
              <a
                aria-current={null}
                className="prime-nav-link"
                href="/results?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                onClick={[Function]}
              >
                Results
              </a>
            </li>
            <li
              className="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden"
            >
              <a
                aria-current={null}
                className="prime-nav-link"
                href="/patients?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                onClick={[Function]}
              >
                People
              </a>
            </li>
            <li
              className="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-user-circle fa-w-16 "
                data-icon="user-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={
                  Object {
                    "fill": "white",
                  }
                }
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </li>
            <li
              className="usa-nav__primary-item usa-sidenav prime-staff-infobox-sidemenu prime-settings-hidden"
            >
              <ul
                className="usa-sidenav__sublist prime-sidenav_inset"
              >
                <li
                  className="usa-sidenav__item span-full-name"
                >
                  Bob Bobberoo
                </li>
                <li
                  className="usa-sidenav__item"
                >
                  <span>
                    <strong>
                      Role: 
                    </strong>
                    Admin user
                  </span>
                </li>
                <li
                  className="usa-sidenav__item"
                >
                  Testing Site
                </li>
              </ul>
            </li>
            <div>
              <div
                className="navlink__support"
              >
                <a
                  href="https://simplereport.gov/support"
                  onClick={[Function]}
                  target="none"
                >
                  Support
                </a>
              </div>
              <button
                className="usa-button usa-button--unstyled"
                onClick={[Function]}
                type="button"
              >
                Log out
              </button>
            </div>
            <li
              className="usa-nav__primary-item prime-settings-hidden"
            >
              <a
                aria-current={null}
                href="/settings?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-cog fa-w-16 "
                  data-icon="cog"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                 Settings
              </a>
            </li>
          </ul>
        </nav>
        <nav
          aria-label="Primary navigation"
          className="usa-nav prime-nav"
        >
          <ul
            className="usa-nav__primary usa-accordion"
          >
            <li
              className="usa-nav__primary-item"
            >
              <a
                aria-current="page"
                className="prime-nav-link active-nav-item"
                href="/queue?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                id="conduct-test-nav-link"
                onClick={[Function]}
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Conduct Tests
              </a>
            </li>
            <li
              className="usa-nav__primary-item"
            >
              <a
                aria-current={null}
                className="prime-nav-link"
                href="/results?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                id="results-nav-link"
                onClick={[Function]}
              >
                Results
              </a>
            </li>
            <li
              className="usa-nav__primary-item"
            >
              <a
                aria-current={null}
                className="prime-nav-link"
                href="/patients?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                id="patient-nav-link"
                onClick={[Function]}
              >
                People
              </a>
            </li>
          </ul>
          <div
            className="prime-facility-select"
          >
            <div
              className="usa-form-group prime-dropdown "
            >
              <select
                aria-required="false"
                className="usa-select"
                id="1"
                onChange={[Function]}
                value="fec4de56-f4cc-4c61-b3d5-76869ca71296"
              >
                <option
                  value="fec4de56-f4cc-4c61-b3d5-76869ca71296"
                >
                  Testing Site
                </option>
              </select>
            </div>
          </div>
          <ul
            className="usa-nav__primary usa-accordion"
          >
            <li
              className="usa-nav__primary-item nav__primary-item-icon"
            >
              <a
                aria-current={null}
                href="/queue#?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-user-circle fa-w-16 "
                  data-icon="user-circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={
                    Object {
                      "color": "",
                    }
                  }
                  viewBox="0 0 496 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </a>
              <div
                aria-label="Primary navigation"
                className="shadow-3 prime-staff-infobox"
              >
                <ul
                  className="usa-sidenav__sublist"
                >
                  <li
                    className="usa-sidenav__item span-full-name"
                  >
                    Bob Bobberoo
                  </li>
                  <li
                    className="usa-sidenav__item"
                  >
                    <span>
                      <strong>
                        Role: 
                      </strong>
                      Admin user
                    </span>
                  </li>
                  <li
                    className="usa-sidenav__item"
                  >
                    Testing Site
                  </li>
                  <li
                    className="usa-sidenav__item navlink__support"
                  >
                    <a
                      href="https://simplereport.gov/support"
                      onClick={[Function]}
                      target="none"
                    >
                      Support
                    </a>
                  </li>
                  <li
                    className="usa-sidenav__item margin-top-2"
                  >
                    <button
                      className="usa-button usa-button--unstyled"
                      onClick={[Function]}
                      type="button"
                    >
                       Log out
                    </button>
                  </li>
                </ul>
              </div>
            </li>
            <li
              className="usa-nav__primary-item nav__primary-item-icon"
            >
              <a
                aria-current={null}
                href="/settings?facility=fec4de56-f4cc-4c61-b3d5-76869ca71296"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-cog fa-w-16 "
                  data-icon="cog"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <main
      className="prime-home"
    >
      <div
        className="grid-container"
      >
        <div
          className="position-relative"
        >
          <div
            className="prime-search-container"
          >
            <form
              className="usa-search usa-search--small prime-search-input display-inline-block"
              role="search"
            >
              <label
                className="usa-sr-only"
                htmlFor="search-field-small"
              >
                Search
              </label>
              <div>
                <input
                  autoComplete="off"
                  className="usa-input"
                  id="search-field-small"
                  name="search"
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Search for a person to start their test"
                  type="search"
                  value=""
                />
                <button
                  className="usa-button"
                  disabled={true}
                  onClick={[Function]}
                  type="submit"
                >
                  <span
                    className="usa-sr-only"
                  >
                    Search
                  </span>
                </button>
              </div>
            </form>
          </div>
        </div>
        <div
          className="grid-container prime-center card-container"
        >
          <div
            className="grid-row"
          >
            <div
              className="usa-card__body"
            >
              <p>
                There are no tests running. Search for a person to start their test
                 
              </p>
            </div>
          </div>
        </div>
      </div>
    </main>
    <div
      className="Toastify"
    />
  </div>
</div>
`;

exports[`App should show error UI 1`] = `
<p>
  Server connection error...
</p>
`;
